evm_loader/lib/src/types/tracer_rocks_db.rs

```diff
@@ -0,0 +1,151 @@
+    account::Account,
+    clock::{Slot, UnixTimestamp},
+    pubkey::Pubkey,
+};
+use std::env;
+use std::str::FromStr;
+use std::sync::Arc;
+use tracing::info;
+
+#[derive(Clone, Serialize)]
+pub struct AccountParams {
+    pub pubkey: Pubkey,
+    pub slot: u64,
+    pub tx_index_in_block: Option<u64>,
+}
+
+use crate::types::tracer_ch_common::{EthSyncStatus, RevisionMap};
+use crate::types::{DbResult, RocksDbConfig, TracerDb};
+// use reconnecting_jsonrpsee_ws_client::{Client, CallRetryPolicy, rpc_params, ExponentialBackoff};
+#[derive(Clone, Debug)]
+pub struct RocksDb {
+    #[allow(dead_code)]
+    url: String,
+    client: Arc<WsClient>,
+}
+
+impl RocksDb {
+    #[must_use]
+    pub async fn new(config: &RocksDbConfig) -> Self {
+        let host = &config.rocksdb_host;
+        let port = &config.rocksdb_port;
+        let url = format!("ws://{host}:{port}");
+
+        // match Client::builder()
+        //     .retry_policy(
+        //     ExponentialBackoff::from_millis(100)
+        //         .max_delay(Duration::from_secs(10))
+        //         .take(3),)
+        match WsClientBuilder::default().build(&url).await {
+            Ok(client) => {
+                let arc_c = Arc::new(client);
+                tracing::info!("Created rocksdb client at {url}");
+                Self { url, client: arc_c }
+            }
+            Err(e) => panic!("Couln't start rocksDb client at {url}: {e}"),
+        }
+    }
+}
+
+#[async_trait]
+impl TracerDb for RocksDb {
+    async fn get_block_time(&self, slot: Slot) -> DbResult<UnixTimestamp> {
+        let response: String = self
+            .client
+            .request("get_block_time", rpc_params![slot])
+            .await?;
+        info!(
+            "get_block_time for slot {:?} response: {:?}",
+            slot, response
+        );
+        Ok(i64::from_str(response.as_str())?)
+    }
+
+    async fn get_earliest_rooted_slot(&self) -> DbResult<u64> {
+        let response: String = self
+            .client
+            .request("get_earliest_rooted_slot", rpc_params![])
+            .await?;
+        info!("get_earliest_rooted_slot response: {:?}", response);
+        Ok(u64::from_str(response.as_str())?)
+    }
+
+    async fn get_latest_block(&self) -> DbResult<u64> {
+        let response: String = self
+            .client
+            .request("get_last_rooted_slot", rpc_params![])
+            .await?;
+        info!("get_latest_block response: {:?}", response);
+        Ok(u64::from_str(response.as_str())?)
+    }
+
+    async fn get_account_at(
+        &self,
+        pubkey: &Pubkey,
+        slot: u64,
+        tx_index_in_block: Option<u64>,
+    ) -> DbResult<Option<Account>> {
+        info!("get_account_at {pubkey:?}, slot: {slot:?}, tx_index: {tx_index_in_block:?}");
+
+        let response: String = self
+            .client
+            .request(
+                "get_account",
+                rpc_params![pubkey.into_owned(), slot, tx_index_in_block],
```

Comment:
Looks like `pubkey` serialized as array of bytes (`{"jsonrpc":"2.0","id":1234,"method":"get_account","params":[[5,120,35,30,206,140,108,201,172,243,104,231,201,68,129,103,101,47,85,1,38,95,115,206,73,53,209,122,159,162,127,239]]}`).
It makes difficults to communicate with the server side. Much better to serialize pubkey as string, so anyone can make RPC request with native ways.
The same about `Signature` field.

Reply:
I fixed it on neon-evm side and also geyser-neon-filter side, so these requests can be made from cmd:
GET TRANSACTION INDEX:

>curl --header "Content-Type: application/json" --data '{"jsonrpc": "2.0", "method": "get_transaction_index", "id": "93", "params": {"signature":"677eKqWzmFjsnvN6WxXfmBJwc8HPNeEFQksxu731LH3z7Au3CWYbhJ3JWazZDpXdtzgUCFut9jpNDTdK2TaAxoyJ"}}' http://localhost:9888
{"jsonrpc":"2.0","result":"2","id":"93"}

GET ACCOUNT:
>curl --header "Content-Type: application/json" --data '{"jsonrpc": "2.0", "method": "get_account", "id": "93", "params": {"pubkey":"SysvarRecentB1ockHashes11111111111111111111", "slot": 12394, "write_version": 0}}' http://localhost:9888